test_queue.py

```diff
@@ -0,0 +1,55 @@
+from queue import Queue
+import pytest
+
+
```

Comment:
You have good test names and well-constructed tests, but I think every test desrves a docstring so you don't have to try and decipher the process from the code as you debug.